src/main/java/com/xceptance/xlt/mastercontroller/TestLoadProfileConfiguration.java

```diff
@@ -477,17 +477,24 @@ else if (LoadFunctionUtils.isComplexLoadFunction(users))
                 complexLoadFunction = users;
             }
 
-            // apply ramp-up parameters to either users or arrival rates
-            if (LoadFunctionUtils.isSimpleLoadFunction(users))
+            // handle ramp-up parameters
+            if (LoadFunctionUtils.isComplexLoadFunction(users) ||
+                (arrivalRate != null && LoadFunctionUtils.isComplexLoadFunction(arrivalRate)))
```

Comment:
IMO testing variable `complexLoadFunction` for being not null is more appropriate (assuming that XLT gets another load driver that can be specified as complex function) and easier to understand (the load function, if any, is already determined here)

Reply:
Indeed, much simpler! Done.